File: home-control/src/main/kotlin/space/termite/homecontrol/rules/NoopRuleEngine.kt
package space.termite.homecontrol.rules

import com.fasterxml.jackson.databind.JsonNode
import jakarta.enterprise.context.ApplicationScoped
import org.jboss.logging.Logger

/**
 * Placeholder [RuleEngine] that never triggers any action.
 *
 * TODO: replace this with a real implementation, e.g. one backed by a database table of
 * `(originTopic, action) -> (destinationTopic, payload)` rules (similar to the previous
 * `homeRules` DynamoDB table), a config file, or an in-memory map for simpler setups.
 */
@ApplicationScoped
class NoopRuleEngine : RuleEngine {

    private val logger = Logger.getLogger(NoopRuleEngine::class.java)

    override fun resolveActions(originTopic: String, payload: JsonNode?): List<TriggeredAction> {
        logger.debugf("No rule engine implementation configured yet; ignoring message from '%s'", originTopic)
        return emptyList()
    }
}

File: home-control/src/rules/noop-rule-engine.ts
import { Injectable, Logger } from '@nestjs/common';
import { RuleEngine } from './rule-engine';
import { TriggeredAction } from './triggered-action';

/**
 * [RuleEngine] with no rules configured yet.
 *
 * TODO: replace with a real implementation backed by a database, config file, etc.
 */
@Injectable()
export class NoopRuleEngine extends RuleEngine {
  private readonly logger = new Logger(NoopRuleEngine.name);

  /** Always resolves to an empty list. */
  async resolveActions(originTopic: string): Promise<TriggeredAction[]> {
    this.logger.debug(
      `No rule engine configured yet; ignoring message from '${originTopic}'`,
    );
    return [];
  }
}
